### PR TITLE
Safe area improvements

### PR DESCRIPTION
**Pull request by ghost**
Deprecates:
global `nui_safeArea`

Adds:
```
view.nui_safeArea.top
view.nui_safeArea.left
view.nui_safeArea.right
view.nui_safeArea.bottom
```

This will result in clearer & more flexible relation mapping logic, since new `nui_safeArea` relations can be used just like any other view relation.

### CODE REVIEW

**Review by artemnovichkov — APPROVED** [review]

File: Sources/Maker.swift [hunk 1]
```diff
@@ -20,41 +20,42 @@ enum HandlerPriority: Int {
 }
 
 public struct SafeArea {}
+@available(*, deprecated, message: "Use `view.safeArea.<type>` instead")
```
Comment (Antowkos) [inline Sources/Maker.swift:23]: It would be better to have line break before this line

File: Sources/MakerHelper.swift [hunk 2]
```diff
@@ -45,16 +45,47 @@ extension Maker {
...
+        case .right:
+            return convertedRect.maxX
+        case .left:
+            return convertedRect.minX
+        case .safeArea(.top):
+            if #available(iOS 11.0, *) {
+                return convertedRect.minY + view.safeAreaInsets.top
+            }
+
+            return convertedRect.minY
```
Comment (Antowkos) [inline Sources/MakerHelper.swift:65]: Not sure if this logic is correct. Before iOS 11 there is `topLayoutGuide`. Can we somehow check this? 🤔

If not we should add docs for that case with detailed description

File: Sources/UIView+Installer.swift [hunk 3]
```diff
@@ -55,60 +55,68 @@ public extension UIView {
...
     ///
     /// - parameter state:          The state for which you configure frame. Default value: `DEFAULT_STATE`.
     /// - parameter installerBlock: The installer block within which you can configure frame relations.
-    
-    public func configureFrame(state: AnyHashable = DEFAULT_STATE, installerBlock: InstallerBlock) {
+
+    func configureFrame(state: AnyHashable = DEFAULT_STATE, installerBlock: InstallerBlock) {
+        guard self.superview != nil else {
+            return
+        }
```
Comment (Antowkos) [inline Sources/UIView+Installer.swift:74]: Thanks for that! 🙏

**Review by Antowkos — CHANGES_REQUESTED** [review]

File: Sources/MakerHelper.swift (hunk 2 above)
Comment (disabled) [reply Sources/MakerHelper.swift:65]: I'm not sure it's a good idea to add any hacks to try to get layout guides here. 
I'd rather just make `nui_safeArea` available only from iOS 11+ if it helps avoid confusion.

Safe areas and layout guides are too different beasts, so it's better not to mix those.
If anyone does layout for iOS < 11, they really should do `maker.top(to: view.nui_top, inset: topLayoutGuide.length)`.

**Review by evgenmikhaylov — APPROVED** [review]

**Review by ReshetovE — APPROVED** [review]

**Review by ntunin — APPROVED** [review]

**Pull request merged**